cyral/resource_cyral_repository_conf_auth.go

```diff
@@ -204,6 +210,13 @@ func repositoryConfAuthResourceSchemaV0() *schema.Resource {
 				Optional:    true,
 				Default:     defaultRepoTLS,
 			},
+			"auth_type": {
+				Description: "Authentication type for this repository. 0 for Opaque Token, " +
+					"1 for AWS IAM. Defaults to 0",
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  defaultAuthType,
+			},
```

Comment:
Lets also add a `ValidateFunc` here to ensure that the enum validation is also done in the terraform provider. I understand that the API already validates that, but doing that directly in the terraform provider enhances the UX by validating during terraform plan instead of during execution, and also gives us flexibility to provide better error messages to users. Adding these kinds of validations in the provider is something that we opted to do in favor of a better UX for customers.
```suggestion
			"auth_type": {
				Description: "Authentication type for this repository. 0 for Opaque Token, " +
					"1 for AWS IAM. Defaults to 0",
				Type:     schema.TypeInt,
				Optional: true,
				Default:  defaultAuthType,
				ValidateFunc: validation.StringInSlice(repoAuthTypes(), false),
			},
```